### ISSUE TITLE
+(a::List...) is precursor for method ambiguity

### ISSUE THREAD

**Issue by rleegates** [MikeInnes/Lazy.jl#40]
...which includes `+()` such that it immediately becomes ambiguous with a definition of `+(a::SomeOtherType...)` somewhere else (in my case I encountered this conflict while using both MultiPoly and Lazy). See the [discussion](https://groups.google.com/forum/#!topic/julia-users/dhOTcnqmcmQ) on julia-users.

**Comment by MikeInnes** [comment]
Yup, good catch! Happy to take a PR to fix this, if you're up for it.
